src/main/scala/za/co/absa/hyperdrive/trigger/scheduler/executors/spark/SparkEmrClusterServiceImpl.scala

```diff
@@ -55,12 +57,28 @@ class SparkEmrClusterServiceImpl @Inject()(sparkConfig: SparkConfig, emrClusterP
       val stepId = response.getStepIds.asScala.headOption
       logger.info(s"Added jobFlowStepsRequest ${jobFlowStepsRequest} for executorId ${id} and stepId $stepId}")
       logger.info(response.toString)
+      stepId
+    }.flatMap { stepId =>
+      val jiStepId = jiSubmitting.copy(stepId = stepId)
+      updateJob(jiStepId)
```

Comment:
If updateJob doesn't pass it will end up with status Lost and will get executed again (2 instances of the same workflow/job).
Is it possible to pass unique identifier into step? Similar to yarn?

Reply:
@jozefbakus I updated the PR, as discussed